packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -508,24 +508,50 @@ async function proxyRequestToAxios(
 	});
 
 	if (configObject.auth?.sendImmediately === false) {
+		// for digest-auth
 		const { auth } = axiosConfig;
 		delete axiosConfig.auth;
-		const header = await axios(axiosConfig).catch((error) => {
+		const resp1 = await axios(axiosConfig).catch((error) => {
 			if (error.response.status === 401) {
-				return error.response.headers['www-authenticate'];
+				return error;
 			}
 			throw error;
 		});
-		const userpass = `${auth?.username as string}:${auth?.password as string}`;
-		axiosConfig.headers = {};
-		axiosConfig.headers.Authorization = digest(
-			axiosConfig.method,
-			axiosConfig.url,
-			header,
-			userpass,
-		);
-		axiosConfig.headers.Authorization += ', algorithm=MD5';
+		const authDetails = resp1.response.headers['www-authenticate']
+			.split(',')
+			.map((v: string) => v.split('='));
+		const nonceCount = `000000001`.slice(-8);
+		const cnonce = crypto.randomBytes(24).toString('hex');
+		const realm: string = authDetails
+			.find((el: any) => el[0].toLowerCase().indexOf('realm') > -1)[1]
+			.replace(/"/g, '');
+		const nonce: string = authDetails
+			.find((el: any) => el[0].toLowerCase().indexOf('nonce') > -1)[1]
+			.replace(/"/g, '');
+		const ha1 = crypto
+			.createHash('md5')
+			.update(`${auth?.username as string}:${realm}:${auth?.password as string}`)
+			.digest('hex');
+		const path = new url.URL(axiosConfig.url!).pathname;
+		const ha2 = crypto
+			.createHash('md5')
+			.update(`${axiosConfig.method ?? 'GET'}:${path}`)
+			.digest('hex');
+		const response = crypto
+			.createHash('md5')
+			.update(`${ha1}:${nonce}:${nonceCount}:${cnonce}:auth:${ha2}`)
+			.digest('hex');
+		const authorization =
+			`Digest username="${auth?.username as string}",realm="${realm}",` +
+			`nonce="${nonce}",uri="${path}",qop="auth",algorithm="MD5",` +
+			`response="${response}",nc="${nonceCount}",cnonce="${cnonce}"`;
```

Comment:
We can move this whole block to a separate function inside the same file that receives the header string, parses it, generates the header and returns it as a string.